machinelearning/llama.cpp.md

```diff
@@ -0,0 +1,73 @@
+# Large Language Model (LLM) inference on Graviton CPUs with llama.cpp
+
+**Introduction**
+
+The main goal of [llama.cpp](https://github.com/ggerganov/llama.cpp) is to enable LLM inference with minimal setup and state-of-the-art performance on a wide variety of hardware. It's a plain C/C++ implementation without any dependencies. It supports quantized general matrix multiply and add (GEMM) kernels for faster inference and reduced memory use. The quantized GEMM kernels are optimized for AWS Graviton processors using Arm Neon and SVE based matrix multiply and add (MMLA) instructions. This document covers how to build and run llama.cpp efficiently for LLM inference on AWS Graviton based Amazon EC2 instances.
+
+# How to use llama.cpp on Graviton CPUs
+
+Building from sources is the recommended way to use llama.cpp on Graviton CPUs, and for other hardware platforms too. This section provides the instructions on how to build it from sources.
+
+**Prerequisites**
+
+It is highly recommended to use the AMIs based on Linux Kernel 5.10 and beyond for the best LLM inference performance on Graviton3 instances. The following queries can be used to list down the AMIs with Kernel 5.10 and beyond.
+
+```
+# For Kernel 5.10 based AMIs list
+aws ec2 describe-images --owners amazon --filters "Name=architecture,Values=arm64" "Name=name,Values=*kernel-5.10*" --query 'sort_by(Images, &CreationDate)[].Name'
+
+# For Kernel 6.x based AMIs list
+aws ec2 describe-images --owners amazon --filters "Name=architecture,Values=arm64" "Name=name,Values=*kernel-6.*" --query 'sort_by(Images, &CreationDate)[].Name'
+```
+
+**Build llama.cpp from source**
+
+```
+# Clone llama.cpp
+git clone https://github.com/ggerganov/llama.cpp
+cd llama.cpp
+
+# build with make
+make
+
+```
+
+# Run LLM inference with llama.cpp
+
+llama.cpp provides a set of tools to (1) convert model binary file into GPT-Generated Unified Format (GGUF) (2) quantize single and half precision format models into one of the quantized formats, for example into Q4_0 (3) rearrange model weights into blocked layout to leverage the hardware specific GEMM kernels and improve the performance and (4) run LLM inference locally.
```

Comment:
I think you need to add some commas, but since you already have nested commas within your list, you can use semicolons instead.

```suggestion
llama.cpp provides a set of tools to (1) convert model binary file into GPT-Generated Unified Format (GGUF); (2) quantize single and half precision format models into one of the quantized formats, for example into Q4_0; (3) rearrange model weights into blocked layout to leverage the hardware specific GEMM kernels and improve the performance; and (4) run LLM inference locally.
```